### ISSUE TITLE
Using constant (non-uniform) N on fixed grid results in very minimal retreat

### ISSUE THREAD

**Issue by glugeorge** [glugeorge/coupled_ice_hydrology#1]
<img width="493" alt="image" src="https://user-images.githubusercontent.com/25989736/199595551-0d389d39-dd02-4e31-8828-2e7cb67183c8.png">
It reaches steady state almost instantly (had to reduce timestep by a lot), and is very close to the initial condition

**Comment by glugeorge** [comment]
<img width="491" alt="image" src="https://user-images.githubusercontent.com/25989736/199599227-cab5d02a-8f81-48ba-aa8c-f99bf803f977.png">
x-axis: distance from divide, y-axis: effective pressure
<br>
N is indeed staying the same (these are N profiles along the channel at t = 0y to t = 25y. Zooming in on terminus:
<img width="485" alt="image" src="https://user-images.githubusercontent.com/25989736/199599441-86e682e2-a3a9-45fb-94ef-a3a83a7693ce.png">

Looks like when N is nonzero at the terminus the ice doesn't like to move? Could be because that means that shear stress is high in that region. 

The steady state N that the ice cannot move past is 4.4267e+04 Pa. (When C = 0.1, perturbing from A = 2.9e-25 to A = 4.9e-25). 

Another observation is that increasing C (making higher friction bed) results in a faster steady state + less retreat. 

Next step will be to look term by term in the equations

**Comment by jkingslake** [comment]
interesting! So this is after you corrected for the $\sigma$ scaling of $N$ to keep $N$ constant everywhere (in terms of real, non-scaled space)?

What do color of the dots mean in the bottom plots?

(on a different topic, feel free to assign me to issues like this, so that I am notified and I have it in a list somewhere).

**Comment by glugeorge** [comment]
@jkingslake Yep, I kept N constant everywhere. The color dots (marking effective pressure at the new terminus) is just for me to discern timesteps - I just plotted the first 5 timesteps with r y g b k colors.

**Comment by jkingslake** [comment]
How large are the time steps?

How far for the 'dynamic $N$' grounding line move in the end?

**Comment by glugeorge** [comment]
Here, the time steps are 5 years each. The dynamic N grounding line moves 38 km whereas the static N grounding line moves 2km.

**Comment by jkingslake** [comment]
OK, very interesting. So its like we thought, the higher $N$ reduces the change. It would be interesting to know how much that depends on the characteristic shape of $N$ which we have been talking about (with the peak 10-20km back from the GL). 

By the way, so axes labels would help with others keeping track of this (and us coming back to this in a year's time ;-)  

Great work though! So cool to see some strange behavour we dont yet understand emerge from the model!

**Comment by glugeorge** [comment]
I've tried switching out the sliding law, such that now we're following a constant*u sliding law, and then I matched the initial steady state non-dimensional basal shear term with it to invert for this constant which I dubbed lambda. This constant will represent the static bed properties. Like the situation where I use the same sliding law but keep N static and constant over time, this model also retreats very minimally. 
Here's what the retreat looked like with the same sliding law, constant N:
![image](https://user-images.githubusercontent.com/25989736/208191637-f80a77e7-cb3e-462e-bdb9-1a6ad13b1654.png)
Here's what it looks like with the new power law, N absorbed into the constant:
<img width="537" alt="image" src="https://user-images.githubusercontent.com/25989736/208191748-09408451-262f-482a-ac22-559f86cb8b3f.png">

Even less retreat!

**Comment by jkingslake** [comment]
Cool. I think this sounds really good, but just to make sure I know exactly what you are doing, you have the maths written out?

**Comment by glugeorge** [comment]
From the nondimensional equation for the ice stress balance, I set the old basal shear term equal to the new law: $\gamma N (\frac{u}{u+N^n})^{1/n} = \lambda u$. I'm claiming that any relevant non-dimensionalization gets absorbed into the $\lambda$. The u scale with A_s from the old law gives a scale of around 300 m/yr which I think is a valid scale to use, and As doesn't appear anywhere else so I'm fine with not creating an entire new law. With those two sides equal, I just plug in the steady state coupled solution for u and N and solve for $\lambda$. I then keep $\lambda$ static and equal and allow u to evolve.
